app/models/queue_item.rb

```diff
@@ -23,18 +31,12 @@ def category_name
 
   private
 
-  def set_position
-    queue = self.user ? self.user.queue_items : [] # FIXME: just to pass the uniqueness shoulda matcher tests!!!
-    max_position = queue.empty? ? 0 : queue.count
-    self.position = max_position + 1
+  def review
+    @review ||= Review.find_by(creator: user, video: video)
   end
 
-  def reassign_positions
-    remaining_queue = QueueItem.all.select do |queue_item|
-      queue_item.user == self.user and queue_item.position > self.position
-    end
-    remaining_queue.each do |queue_item|
-      queue_item.update(position: queue_item.position - 1)
-    end
+  def set_position
+    queue = QueueItem.where(user: user)
+    self.position ||= queue.count + 1
```

Comment:
OK 1 more thing. I realised i told you 1 wrong information. `user` is not the column in `queue_items` it is because of `belongs_to :user` as that `belongs_to` specifies methods and one of them is `.user` which you use in that case when you call `user`. Sorry if I confused you.